New artifact for github action webhooks

robertjahn opened the pull request.

robertjahn commented:
need feedback for the following:
* any requirement on version or just use 0.1.0 ?
* OK with the names I used?
* where should I get image URL for github? can I just put an image in the image subfolder of this repo and point to it
* is the overall too much detail?

johannes-b commented:
@robertjahn can you please sign your commits using the commands provided here: https://github.com/keptn-sandbox/artifacthub/pull/34/checks?check_run_id=4262603891

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+# GitHub Actions via Keptn Webhooks
```
Please change to: 

`# GitHub Actions Webhook Integration`

robertjahn replied:
fixed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/a0ab4afe44744d3ee98494d4f368c60a5f18bea4

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+
+This integration shows how to invoke GitHub Action workflows leveraging [Keptn's Webhook Service](https://keptn.sh/docs/0.10.x/integrations/webhooks/).  With a GitHub actions integration, you can call existing GitHub workflows from Keptn sequences. 
+
+[WATCH THIS VIDEO TUTORIAL](https://www.youtube.com) to see this in action!
```
Link to video is missing.

robertjahn replied:
Have an ask out to Keptn team for this

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+
+<img src="images/createsecret.png" width="50%" height="50%">
+
+## Step 3: Create Keptn project
```
Please remove "Step 3" as we can (should) assume that a user knows that. 

And I would like to avoid writing a full tutorial here.

robertjahn replied:
removed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/a0ab4afe44744d3ee98494d4f368c60a5f18bea4

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+There are two use cases for Keptn webhooks:
+
+1. **Silent** - The webhook subscription listens for status events, `started` or `finished`, and calls the GitHub Action API to invoke the GitHub workflow. GitHub does not need to send anything back to Keptn and the Keptn sequence will just continue to process.
+1. **Interactive** -  The webhook subscription listens for`triggered` events, calls the GitHub Action API to invoke the GitHub workflow. Once it does the  Keptn sequence will wait until the GitHub workflow to send back a `finished` event.
```
space between: for`triggered`

robertjahn replied:
fixed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/a0ab4afe44744d3ee98494d4f368c60a5f18bea4

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+
+## Use Case 1: Silent webhook
+
+This example shows a GitHub workflow that is triggered when a Keptn task called `myTask` sends its `finished` Keptn event within the `dev` or `staging` stage of the Keptn project. 
```
Below you use the example of the `evaluation` task. Hence, change `myTask` to `evaluation`

robertjahn replied:
fixed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/a0ab4afe44744d3ee98494d4f368c60a5f18bea4

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,257 @@
+1. In order for the GitHub workflow to call the Keptn API, create two [GitHub repository action secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) with the Keptn API URL and Keptn API Token.
+
+    * Go to GitHub repos `Settings > Secret` and click add these two secrets:
+      * `KEPTN_BASE_URL` = Use the base URL for your environment such as: `https://[YOUR-TENANT].cloudautomation.dynatrace.com` for the secret value.
```
Please change `cloudautomation.dynatrace.com` to a `demo.keptn.com`

robertjahn replied:
fixed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/2e1e852a46742a14fb2e32877d27e6bf5debf8ae

johannes-b commented on githubaction/0.1.0/README.md:
````diff
@@ -0,0 +1,257 @@
+            curl -X POST "${{ secrets.KEPTN_API_URL }}/v1/event" -H "Content-Type: application/json" -H "accept: application/json" -H "x-token: ${{ secrets.KEPTN_API_TOKEN }}" -d "$json"
+    ```
+
+1. With those steps done, Keptn will trigger a GitHub Action whenever an `mysequence.triggered` event occurs.  To test this example setup:
````
Change `mysequence.triggered` to `mytask-interactive.triggered`

johannes-b commented on githubaction/0.1.0/artifacthub-pkg.yml:
```diff
@@ -0,0 +1,27 @@
+displayName: GitHub Action Webhook Integration
+createdAt: 2021-11-17T00:00:00Z
+description: This integration shows how to invoke GitHub Action workflows leveraging Keptn's webhook service.
+logoURL: https://upload.wikimedia.org/wikipedia/commons/thumb/b/b9/Slack_Technologies_Logo.svg/2560px-Slack_Technologies_Logo.svg.png
```
Please change logo to a GitHub Actions logo

robertjahn replied:
Yes but  should I put an image in this repo? or does it need to be an external  URL one as was done in the slack example?

johannes-b replied:
I propose to add an image to `/images` and the reference it using: `https://raw.githubusercontent.com/keptn-sandbox/artifacthub/main/slack/0.1.0/images/xyz.png`

johannes-b commented on githubaction/0.1.0/artifacthub-pkg.yml:
```diff
@@ -0,0 +1,27 @@
+keywords:
+  - keptn
+  - webhook
+  - github
```
please add `githubactions`

robertjahn replied:
fixed in https://github.com/keptn-sandbox/artifacthub/pull/34/commits/a0ab4afe44744d3ee98494d4f368c60a5f18bea4

johannes-b commented on githubaction/0.1.0/artifacthub-pkg.yml:
```diff
@@ -0,0 +1,27 @@
+  - name: Source
+    url: https://keptn.sh/docs/0.10.x/integrations/webhooks/
+  - name: Video
+    url: https://www.youtube.com
```
Please add the youtube link to your recording.

robertjahn commented:
> * any requirement on version or just use 0.1.0 ?

I was thinking maybe it should be 1.0.0 instead.

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,242 @@
+
+The completed form should look as follows:
+
+<img src="images/createsecret.png" width="50%" height="50%">
```
Please use here an absolute reference like: 

`![](https://raw.githubusercontent.com/keptn-sandbox/artifacthub/main/slack/0.1.0/images/xyz.png`

johannes-b commented on githubaction/0.1.0/README.md:
````diff
@@ -0,0 +1,242 @@
+      ```
+    * The completed form should look as follows:
+    
+      <img src="images/silent-subscription.png" width="50%" height="50%">
````
Please use here an absolute reference like: 

`![](https://raw.githubusercontent.com/keptn-sandbox/artifacthub/main/slack/0.1.0/images/xyz.png`

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,242 @@
+
+    * The completed form should look as follows:
+
+      <img src="images/interactive-subscription.png" width="50%" height="50%">
```
Please use here an absolute reference like: 

`![](https://raw.githubusercontent.com/keptn-sandbox/artifacthub/main/slack/0.1.0/images/xyz.png`

johannes-b commented on githubaction/0.1.0/README.md:
```diff
@@ -0,0 +1,242 @@
+      * `KEPTN_API_TOKEN` = From the Bridge UI, click on the profile icon on the top right. From the profile popup window, copy the API token and use that for the secret value
+    * The completed secrets should look as follows:
+
+      <img src="images/github-secret.png" width="50%" height="50%">
```
Please use here an absolute reference like: 

`![](https://raw.githubusercontent.com/keptn-sandbox/artifacthub/main/slack/0.1.0/images/xyz.png`

johannes-b commented on githubaction/0.1.0/artifacthub-pkg.yml:
```diff
@@ -0,0 +1,28 @@
+  - url: https://artifacthub.io/packages/helm/keptn/keptn
+annotations:
+  keptn/kind: "notification"
+  keptn/org: "sandbox"
```
Please change to `keptn/org: "keptn"`

Because the `webhook-service` lives in `keptn` organization.

johannes-b replied:
@robertjahn please change line 27 just to: `keptn/org: "keptn"`

robertjahn commented:
I am not clear on the update to make for this comment.
https://github.com/keptn-sandbox/artifacthub/pull/34#discussion_r754412610

robertjahn commented:
> need feedback for the following:
> 
> * any requirement on version or just use 0.1.0 ?  

After talking with keptn team will use 0.1.0

> * OK with the names I used?
After talking with keptn team - yes ok with name but have short name in pkg file

> * where should I get image URL for github? can I just put an image in the image subfolder of this repo and point to it

After talking with keptn team using static links to this repo

> * is the overall too much detail?

After talking with keptn team, removed a little.

robertjahn commented:
@johannes-b  -- I "resolved" all conversations.  Please take a final look.  I added youtube link, adjusted image URLS, resized the images a little smaller, make the edits to the PKG file.

robertjahn commented:
> I am not clear on the update to make for this comment. [#34 (comment)](https://github.com/keptn-sandbox/artifacthub/pull/34#discussion_r754412610)

After talking with keptn team adjusted to `keptn/org: "keptn"`

johannes-b reviewed (approved):
lgtm
